aws/data_source_aws_instance.go

```diff
@@ -389,6 +401,13 @@ func instanceDescriptionAttributes(d *schema.ResourceData, instance *ec2.Instanc
 			d.Set("user_data", userDataHashSum(*attr.UserData.Value))
 		}
 	}
+	{
+		creditSpecifications, err := getCreditSpecifications(conn, d.Id())
+		if err != nil {
+			return err
+		}
+		d.Set("credit_specification", creditSpecifications)
```

Comment:
We should return errors when attempting to set non-scalar attributes in the state (otherwise it will silently use the value of the configuration or in the case of the data source, not set it at all):

```go
if err := d.Set("credit_specification", creditSpecifications); err != nil {
  return fmt.Errorf("error setting credit_specification: %s", err)
}
```